plugins/generator-1.19.4/forge-1.19.4/templates/gui/gui_window.java.ftl

```diff
@@ -0,0 +1,277 @@
+ # but WITHOUT ANY WARRANTY; without even the implied warranty of
+ # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ # GNU General Public License for more details.
+ # 
+ # You should have received a copy of the GNU General Public License
+ # along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ # 
+ # Additional permission for code generator templates (*.ftl files)
+ # 
+ # As a special exception, you may create a larger work that contains part or 
+ # all of the MCreator code generator templates (*.ftl files) and distribute 
+ # that work under terms of your choice, so long as that work isn't itself a 
+ # template for code generation. Alternatively, if you modify or redistribute 
+ # the template itself, you may (at your option) remove this special exception, 
+ # which will cause the template and the resulting code generator output files 
+ # to be licensed under the GNU General Public License without this special 
+ # exception.
+-->
+
+<#-- @formatter:off -->
+<#include "../procedures.java.ftl">
+package ${package}.client.gui;
+
+<#assign mx = data.W - data.width>
+<#assign my = data.H - data.height>
+
+public class ${name}Screen extends AbstractContainerScreen<${name}Menu> {
+
+	private final static HashMap<String, Object> guistate = ${name}Menu.guistate;
+
+	private final Level world;
+	private final int x, y, z;
+	private final Player entity;
+
+	<#list data.getComponentsOfType("TextField") as component>
+		EditBox ${component.getName()};
+	</#list>
+
+	<#list data.getComponentsOfType("Checkbox") as component>
+		Checkbox ${component.getName()};
+	</#list>
+
+	<#list data.getComponentsOfType("Button") as component>
+		Button ${component.getName()};
+	</#list>
+
+	<#list data.getComponentsOfType("ImageButton") as component>
+		ImageButton ${component.getName()};
+	</#list>
+
+	public ${name}Screen(${name}Menu container, Inventory inventory, Component text) {
+		super(container, inventory, text);
+		this.world = container.world;
+		this.x = container.x;
+		this.y = container.y;
+		this.z = container.z;
+		this.entity = container.entity;
+		this.imageWidth = ${data.width};
+		this.imageHeight = ${data.height};
+	}
+
+	<#if data.doesPauseGame>
+		@Override public boolean isPauseScreen() {
+			return true;
+		}
+	</#if>
+
+	<#if data.renderBgLayer>
+		private static final ResourceLocation texture = new ResourceLocation("${modid}:textures/screens/${registryname}.png" );
+	</#if>
+
+	@Override public void render(PoseStack ms, int mouseX, int mouseY, float partialTicks) {
+		this.renderBackground(ms);
+		if (!this.getChildAt(mouseX, mouseY).equals(Optional.empty()) ? (this.getChildAt(mouseX, mouseY).get() instanceof Button button && button.isFocused()) || (this.getChildAt(mouseX, mouseY).get() instanceof ImageButton imageButton && imageButton.isFocused()) || (this.getChildAt (mouseX, mouseY).get() instanceof Checkbox checkbox && checkbox.isFocused()) : false)

```

Comment:
I know you explained the need for this hack, but I am quite confident that there must be a better way to do it. 

Do vanilla GUIs also handle this that way?

Reply:
It seems vanilla GUIs don't handle anything at all anymore and just leave widgets focused now, you'll see that happen in the structure block's GUI, the buttons stay focused unless you press another widget.

Reply:
This might just be a bug since it happens for the crafting menu too and looks very off
![image](https://user-images.githubusercontent.com/112079900/227967148-a76918d8-b93d-445d-9fef-2434d7a6b534.png)

Reply:
hm... maybe a new parameter in the GUI element for 1.19.4+ to enable/disable this option?

Reply:
I don't think it is needed, this is how most normal UIs behave. 

One could theoretically make a plugin for this, but if UI behaves like this now, it should for mods too. I don't see a need for this in core

Reply:
alright, I'll get rid of the hack